File: GeneratedEntityReaderClasses/Components/DrugEffectModifierComponent.cs
namespace GeneratedNoitaClasses;
public class DrugEffectModifierComponent : NoitaComponentBase,  iNoitaType<DrugEffectModifierComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public ConfigDrugFx fx_add { get; set; } = new();
	public ConfigDrugFx fx_multiply { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		fx_add.Read(s);
		fx_multiply.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		fx_add.Write(s);
		fx_multiply.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/DrugEffectModifierComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from GeneratedClasses.ConfigDrugFx import ConfigDrugFx

from NoitaStream import NoitaStream
@dataclass
class DrugEffectModifierComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    fx_add:ConfigDrugFx = field(default_factory=ConfigDrugFx)
    fx_multiply:ConfigDrugFx = field(default_factory=ConfigDrugFx)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.fx_add.Read(s)
        self.fx_multiply.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.fx_add.Write(s)
        self.fx_multiply.Write(s)

